File: java/leetcode_grind/Day663.java
package leetcode_grind;

public class Day663 {
    public static class ListNode {
        int val;
        ListNode next;

        ListNode() {
        }

        ListNode(int val) {
            this.val = val;
        }

        ListNode(int val, ListNode next) {
            this.val = val;
            this.next = next;
        }
    }

    // https://leetcode.com/problems/insert-greatest-common-divisors-in-linked-list/description/?envType=daily-question&envId=2024-09-10
    static class Solution1 {
        int gcd(int a, int b) {
            return b > 0 ? gcd(b, a % b) : a;
        }

        public ListNode insertGreatestCommonDivisors(ListNode head) {
            ListNode l = head, r = head.next;
            while (r != null) {
                ListNode m = new ListNode(gcd(l.val, r.val));
                l.next = m;
                m.next = r;
                l = r;
                r = r.next;
            }
            return head;
        }
    }
}

File: src/leetcode_grind/day_663.rs
// https://leetcode.com/problems/minimum-bit-flips-to-convert-number/description/?envType=daily-question&envId=2024-09-11
pub fn min_bit_flips(start: i32, goal: i32) -> i32 {
    let mut ans = 0;
    for i in 0..31 {
        if ((start >> i) & 1) != ((goal >> i) & 1) {
            ans += 1;
        }
    }
    ans
}
